lib/linguist/heuristics.rb

```diff
@@ -61,9 +61,6 @@ def call(data)
       @heuristic.call(data)
     end
 
-    # Common heuristics
-    ObjectiveCRegex = /^[ \t]*@(interface|class|protocol|property|end|synchronised|selector|implementation)\b/
```

Comment:
It much more difficult to differentiate Objective-C from C and C++ than it is to differentiate it from Limbo, M, MUF, Mathematica, Matlab and Mercury. That's why I think we can use a simpler regex (faster?) for the second case.